polyfill/lib/duration.mjs

```diff
@@ -26,31 +26,58 @@ export class Duration {
     seconds = 0,
     milliseconds = 0,
     microseconds = 0,
-    nanoseconds = 0
+    nanoseconds = 0,
+    disambiguation = 'constrain'
   ) {
-    let tdays;
-    ({
-      days: tdays,
-      hour: hours,
-      minute: minutes,
-      second: seconds,
-      millisecond: milliseconds,
-      microsecond: microseconds,
-      nanosecond: nanoseconds
-    } = ES.BalanceTime(
-      ES.AssertPositiveInteger(ES.ToInteger(hours)),
-      ES.AssertPositiveInteger(ES.ToInteger(minutes)),
-      ES.AssertPositiveInteger(ES.ToInteger(seconds)),
-      ES.AssertPositiveInteger(ES.ToInteger(milliseconds)),
-      ES.AssertPositiveInteger(ES.ToInteger(microseconds)),
-      ES.AssertPositiveInteger(ES.ToInteger(nanoseconds))
-    ));
-    days += tdays;
+    years = ES.ToInteger(years);
+    months = ES.ToInteger(months);
+    days = ES.ToInteger(days);
+    hours = ES.ToInteger(hours);
+    minutes = ES.ToInteger(minutes);
+    seconds = ES.ToInteger(seconds);
+    milliseconds = ES.ToInteger(milliseconds);
+    microseconds = ES.ToInteger(microseconds);
+    nanonseconds = ES.ToInteger(nanonseconds);
+
+    switch (disambiguation) {
```

Comment:
We need an `ES.ToString()` call somewhere around here.

Reply:
Done!